tests/form.js

```diff
@@ -111,40 +110,5 @@ define([
             chai.expect(label.p.relevantAttr).to.include("/data/hidden");
             chai.expect(label.p.labelItextID.defaultValue()).to.include("/data/hidden");
         });
-
-        it("should merge data-only nodes with control nodes", function () {
-            var form = new Form({}),
-                values = [];
-
-            form.dataTree = makeTree("data", ["a", "x1", "b", "x2", "c"]);
-            form.controlTree = makeTree("control", ["a", "b", "c"]);
-
-            form.mergedTreeMap(function (v) { values.push(v.id); });
-            assert.equal(values.join(" "), "a x1 b x2 c");
-        });
-
-        it("should prefer control tree order on merge", function () {
-            var form = new Form({}),
-                values = [];
-
-            form.dataTree = makeTree("data", ["a", "x1", "b", "x2", "c"]);
-            form.controlTree = makeTree("control", ["a", "c", "b"]);
-
-            form.mergedTreeMap(function (v) { values.push(v.id); });
-            assert.equal(values.join(" "), "a c b x1 x2");
-        });
```

Comment:
Ha! I knew there were tests for this. These should be converted to XForm with equivalent structure and write tests to verify that the parser preserves order as specified here.